### ISSUE TITLE
Rubocop does not detect typos in Ruby code.

### ISSUE THREAD

**Issue by arjunmenon** [rubocop-hq/rubocop#6011]
Hey
Rubocop does not detect typos in Ruby code.
I had an exception block
```ruby
begin
    #my code
resuce Mechanize::ResponseCodeError => e
    puts "there was an error"
end
```

For some reason it was't working. I tried running rubocop and this was the response

```ruby
lib/tasks/news.rake:28:1: C: Layout/Tab: Tab detected.
	resuce Mechanize::ResponseCodeError => e
^
lib/tasks/news.rake:28:2: C: Layout/IndentationConsistency: Inconsistent indentation detected.
	resuce Mechanize::ResponseCodeError => e
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

It was only 15 min later I realised I had a typo in the block, it was `rescue` instead of `resuce`.

Shouldn't Rubocop be able to detect them as well?

--------

```
$ rubocop -V
0.57.2 (using Parser 2.5.1.0, running on ruby 2.3.3 x86_64-linux)
```

**Comment by benhutton** [comment]
@arjunmenon I believe that this is valid ruby code:

```
def resuce(some_hash)
  puts some_hash.keys.first
  puts some_hash.values.first
end

begin
  resuce(Mechanize::ResponseCodeError => e)
  puts "there was an error"
end
```

Probably not rubocop's job to catch that. And I'd imagine you were getting `NoMethodError` about  `resuce`, right?

**Comment by arjunmenon** [comment]
I had put an exception block.
And Ruby did not report any error about the typo, it just ran and printed the exception, skipping the rescue part, which I had misspelled.

**Comment by koic** [comment]
As @benhutton showed, the theme code is valid Ruby code. Also, `begin` ... `end` is not exclusive to exception handling. It is used in loop processing, for example.

```ruby
i = 0

begin
  puts i += 1
end while i < 5
```

I think that detection of typo for reserved words is out of the scope of RuboCop. This may be a scope of syntax highlighter, or some different tool similar to did_you_mean gem (Unfortunately I don't know such a gem) .

**Comment by bbatsov** [comment]
Yeah, I agree with @koic. It's really hard to find such cases where something produces valid code, but there's a typo in a keyword or a common method. Other than some cop that has a dictionary of all some keywords/identifiers and common typos I don't think there's anything else we can do about this. I recall someone suggested to work on this in the past, but that never happened.
